File: 30 Days of Code/Day 8 Dictionaries and Maps/Solution.cs
using System;
using System.Collections.Generic;
using System.IO;
class Solution {
    static string ReadName( ) {
        string name = Console.ReadLine( ) ;
        
        return name;
    }
    
    static void Main(String[] args) {
        IDictionary< string, int > phoneBook = new Dictionary< string, int >( );
        int n = Int32.Parse( Console.ReadLine( ) );
        
        for( int idx = 1; idx <= n; idx++ ) {
            string[ ] line = Console.ReadLine( ).Split( ' ' );
            phoneBook.Add( line[ 0 ], Int32.Parse( line[ 1 ] ) );    
        }
        
        string name = ReadName( );
        int phone;
        while( !String.IsNullOrEmpty( name ) ) {
            if( phoneBook.TryGetValue( name, out phone ) ) {
                Console.WriteLine( $"{name}={phone}" );
            } else {
                Console.WriteLine( "Not found" );
            }
            name = ReadName( );
        }
    }
}

File: 30 Days of Code/Day 8 Dictionaries and Maps/Solution.py
import sys

n = int( input( ).strip( ) )
phoneBook = { }

for _ in range( n ):
    contact = input( ).split( ' ' )
    phoneBook[ contact[ 0 ] ] = int( contact[ 1 ] )

while True:
    name = input( ).strip( )
    if len( name ) == 0:
        break
    
    if name in phoneBook:
        print( name + "=" + str( phoneBook[ name ] ) )
    else:
        print( "Not found" )
